compiler_opt/distributed/local/local_worker_manager.py

```diff
@@ -189,20 +189,23 @@ def remote_call(*args, **kwargs):
           if self._is_stopped():
             result_future.set_exception(concurrent.futures.CancelledError())
           else:
-            self._send.put(
-                Task(
-                    msgid=msgid,
-                    func_name=name,
-                    args=args,
-                    kwargs=kwargs,
-                    is_urgent=cls.is_priority_method(name)))
+            with self._pipe_lock:
+              self._pipe.send(
+                  Task(
+                      msgid=msgid,
+                      func_name=name,
+                      args=args,
+                      kwargs=kwargs,
+                      is_urgent=cls.is_priority_method(name)))
             self._map[msgid] = result_future
         return result_future
 
       return remote_call
 
     def shutdown(self):
       try:
+        # Killing the process triggers observer death, which triggers msg_pump
+        # death
```

Comment:
nit: msg_pump _exit_ (because 'death' may be taken as forceful exit, but that's not the case)